docs/src/tutorials/manifold.md

```diff
@@ -0,0 +1,198 @@
+This tutorial demonstrated, that you can get your first own manifold quite fast and you only have to implement the functions you actually need. For this tutorial it would be helpful, if you take a look at our [notation](../notation.md).
+For this tutorial it is helpful, if you heard of the exponential map, tangent vectors and the dimension of a manifold.
+
+We start with two technical preliminaries. If you want to start directly, you can [skip](@ref manifold-tutorial-task) this paragraph and revisit it for two of the implementation details.
+
+After that, we will
+
+* [model](@ref manifold-tutorial-task) the manifold
+* [implement](@ref manifold-tutorial-checks) two tests, so that points and tangent vectors can be checked for validity, for example also within [`ValidationManifold`](@ref),
+* [implement](@ref manifold-tutorial-fn) two functions, the exponential map and the manifold dimension.
+
+## [Technical preliminaries](@id manifold-tutorial-prel)
+
+There is only two small  technical things we require.
+First of all our [`Manifold`](@ref)`{𝔽}` has a parameter `𝔽`.
+This parameter indicates the [`number_system`](@ref) the manifold is based on, e.g. real-valued (`ℝ`) or complex-valued (`ℂ`) data.
+This can be used to simultaneously implement both a real-valued and a complex-valued type of a manifold.
+
+Second, a main design decision of `Manifold.jl` is, that most functions are implemented by mutation functions, i.e. as in-place-computations. There usually exists a non-mutating version that falls back to allocating memory and calling the mutating one. This means you only have to implement the mutating version, _unless_ there is a good reason to provide a special case for the non-mutating one, i.e. because in that case you know a far better performing implementation.
+
+Let's look at this for an example. The exponential map $\exp_p\colon T_p\mathcal M \to \mathcal M$ that maps a tangent vector $X\in T_p\mathcal M$ from the tangent space at $p\in \mathcal M$ to the manifold.
+The function [`exp`](@ref exp(M::Manifold, p, X)) has to know the manifold `M`, the point `p` and the tangent vector `X` as input, so to compute the resulting point `q` would be calling
+
+```julia
+q = exp(M, p, X)
+```
+
+If you already have allocated memory to the variable that should store the result, it is better to perform the computations directly in that memory and avoid reallocations. For example
+
+```julia
+q = similar(p)
+exp!(M, q, p, X)
+```
+
+calls [`exp!`](@ref exp!(M::Manifold, q, p, X)), which modifies its input `q` and returns the resulting point in there.
+Actually these two lines are the default implementation for [`exp`](@ref exp(M::Manifold, p, X)).
+Note that for a unified interface, the manifold `M` is _always_ the first parameter, and the variable the result will be stored to in the mutating variants is _always_ the second parameter.
```

Comment:
There will be one exception to this rule, specifically when a function takes another function as an argument, and it would make sense to write that function using the do-notation. Then `M` would be second and the mutating variable would be third.

Reply:
But currently we don't have that.

Reply:
Yes, we don't have that yet on the master branch but it's already a part of #110 . We can of course modify tutorial in that PR then.

Reply:
Maybe changing it, when necessary might be best. And it stays a mostly or usually, still.